lib/auth/permissions.go

```diff
@@ -178,6 +179,17 @@ func (a *authorizer) Authorize(ctx context.Context) (*Context, error) {
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
+	if authContext.Checker.PinSourceIP() {
```

Comment:
This differs between SSH and TLS - SSH certificates validate IPs if `pin_source_ip` was set during certificate creation, TLS certificates validate IPs if `pin_source_ip` is set when user tries to use the certificate. We can modify it so both ways work the same by storing state `pin_source_ip` in custom extension and checking that in the line above.